bokehjs/src/coffee/models/transforms/step_interpolator.coffee

```diff
@@ -0,0 +1,50 @@
+_ = require "underscore"
+Transform = require "./transform"
+Interpolator = require "./interpolator"
+p = require "../../core/properties"
+
+
+class StepInterpolator extends Interpolator.Model
+
+  @define {
+      mode: [ p.TransformStepMode, "after"]
+    }
+
+  compute: (x) ->
+    # Apply the transform to a single value
+    @sort(descending = false)
+
+    ind = -1
+    if @get('mode') == "after"
+      ind = _.findLastIndex(@_x_sorted, (num) ->
+        return x >= num
+      )
+
+    if @get('mode') == "before"
+      ind = _.findIndex(@_x_sorted, (num) ->
+        return x <= num
+      )
+
+    if @get('mode') == "center"
+      diffs = (Math.abs(tx - x) for tx in @_x_sorted)
+      mdiff = _.min(diffs)
+      ind = _.findIndex(diffs, (num) ->
+        return mdiff == num
+      )
+
+    if ind != -1
+      ret = @_y_sorted[ind]
+    else
+      ret = NULL
```

Comment:
WIP :)

> On Apr 13, 2016, at 10:39 AM, Mateusz Paprocki <notifications@github.com> wrote:
> 
> In bokehjs/src/coffee/models/transforms/step_interpolator.coffee:
> 
> > +    if @get('mode') == "before"
> > +      ind = _.findIndex(@_x_sorted, (num) ->
> > +        return x <= num
> > +      )
> > +
> > +    if @get('mode') == "center"
> > +      diffs = (Math.abs(tx - x) for tx in @_x_sorted)
> > +      mdiff = _.min(diffs)
> > +      ind = _.findIndex(diffs, (num) ->
> > +        return mdiff == num
> > +      )
> > +
> > +    if ind != -1
> > +      ret = @_y_sorted[ind]
> > +    else
> > +      ret = NULL
> null?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>